### PR TITLE
[DRAFT] SAIC-768: Volume image metadata is not migrated

### PR DESCRIPTION
**Pull request by aguzikova**
Add volume_glance_metadata table migrating with replacing old (SRC)
image ids by new ones (DST).

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1735/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:2: [E0401(import-error), ] Unable to import 'test_exceptions'
devlab/tests/test_resource_migration.py:355: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:365: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:2: [E0401(import-error), ] Unable to import 'test_exceptions'
devlab/tests/test_resource_migration.py:355: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:365: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/actions/cinder_database_manipulation.py [hunk 1]
```diff
@@ -173,6 +175,76 @@ def __init__(self, *args, **kwargs):
...
+                            img['image'][f] != dst_img['image'][f]:
+                        break
+                    if f == 'owner' and \
+                            keystone[SRC].try_get_tenant_name_by_id(
+                                img['image'][f]) != \
+                            keystone[DST].try_get_tenant_name_by_id(
+                                dst_img['image'][f]):
+                        break
+                else:
+                    self.img_map[img['image']['id']] = dst_img['image']['id']
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: Please move this logic to glance and rename to something like `glance_images_identical(src_image, dst_image)`

**Comment by MirantisWorkloadMobility** [comment]
$PULL_STATUS
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1784/

**Comment by MirantisWorkloadMobility** [comment]
$PYLINT_STATUS

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.base.migration
cloudferrylib/base/migration.py:45: [E1101(no-member), Migration.get_resource] Instance of 'Migration' has no 'resource' member
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:1: [E0401(import-error), ] Unable to import 'config'
devlab/tests/test_resource_migration.py:2: [E0401(import-error), ] Unable to import 'test_exceptions'
devlab/tests/test_resource_migration.py:3: [E0401(import-error), ] Unable to import 'functional_test'
devlab/tests/test_resource_migration.py:355: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:365: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1786/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.base.migration
cloudferrylib/base/migration.py:45: [E1101(no-member), Migration.get_resource] Instance of 'Migration' has no 'resource' member
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:1: [E0401(import-error), ] Unable to import 'config'
devlab/tests/test_resource_migration.py:2: [E0401(import-error), ] Unable to import 'test_exceptions'
devlab/tests/test_resource_migration.py:3: [E0401(import-error), ] Unable to import 'functional_test'
devlab/tests/test_resource_migration.py:355: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:365: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class
************* Module tests.cloudferrylib.os.actions.test_cinder_database_manipulation
tests/cloudferrylib/os/actions/test_cinder_database_manipulation.py:187: [W0613(unused-argument), _action.not_rsync] Unused argument 'src_size'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:1: [E0401(import-error), ] Unable to import 'config'
devlab/tests/test_resource_migration.py:2: [E0401(import-error), ] Unable to import 'test_exceptions'
devlab/tests/test_resource_migration.py:3: [E0401(import-error), ] Unable to import 'functional_test'
devlab/tests/test_resource_migration.py:355: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:365: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/base/migration.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+
+from cloudferrylib.utils import utils
+
+
+CLOUD = 'cloud'
+SRC, DST = 'src', 'dst'
+
+
+class Migration(object):
```
Comment (roman-verchikov) [inline cloudferrylib/base/migration.py]: This seem to only be used in cinder migration, doesn't need to be moved to a `base` package, it's not globally base

File: cloudferrylib/base/migration.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                           resource_type=resource_type)
+
+    def obj_identical(self, src, dst):
+        """Compare src and dst objects from resource info.
+
+        :return: boolean
+
+        """
+
+        ignore_fields = ['id', 'resource', 'owner']
```
Comment (roman-verchikov) [inline cloudferrylib/base/migration.py]: Comparison for object equity should be done the other way around, it has to be done over explicit set of fields, not implicit, e.g. images are considered identical if name, tenant and md5 match.

File: cloudferrylib/base/migration.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+        :return: boolean
+
+        """
+        ignore_fields = ['id', 'description']
+
+        for field in src:
+            if field not in ignore_fields and \
+                    src[field] != dst[field]:
+                return False
```
Comment (roman-verchikov) [inline cloudferrylib/base/migration.py]: Comparison for object equity should be done the other way around, it has to be done over explicit set of fields, not implicit, e.g. images are considered identical if name, tenant and md5 match

**Comment by roman-verchikov** [comment]
Let's keep this on hold until we clarify the importance of metadata migration

**Pull request closed without merging**